auto_dev/workflow_manager.py

```diff
@@ -246,17 +248,85 @@ def to_yaml(self):
     @staticmethod
     def from_yaml(file_path: str):
         """Load the workflow manager from yaml."""
-        raw_data = WorkflowManager.load_yaml(file_path)
+        raw_data = IO.load_data(file_path)
         raw_data["tasks"] = [Task(**task) for task in raw_data["tasks"]]
         wf = WorkflowManager()
         wf.add_workflow(Workflow(**raw_data))
         return wf
 
     @staticmethod
-    def load_yaml(file_path: str):
-        """Load a yaml file."""
-        with open(file_path, encoding="utf-8") as file:
-            return yaml.safe_load(file)
+    def replace_placeholders(obj: Any, key_value_pairs: dict[str, Any]) -> Any:
+        """Replace placeholders in any data structure with provided key-value pairs.
+
+        Args:
+        ----
+            obj: The object to process (can be dict, list, str, or other types)
+            key_value_pairs: Dictionary of key-value pairs to use for replacement
+
+        Returns:
+        -------
+            The processed object with all placeholders replaced
+
+        """
+        if isinstance(obj, dict):
+            return {k: WorkflowManager.replace_placeholders(v, key_value_pairs) for k, v in obj.items()}
```

Comment:
so we need to re-architect this,

A) we need to have a kwarsg key on the workflow definition

B) we access the kwargs the same syntax we use for access the output of the previous steps.

C) then we can just update the workflow with the kwarrrgs object making for much much easier imputation and running.